backend/services/vector_store.py

```
@@ -0,0 +1,501 @@
+            # Truncate or pad to 1536 dimensions
+            if len(embedding) > 1536:
+                embedding = embedding[:1536]
+            else:
+                embedding = embedding + [0.0] * (1536 - len(embedding))
+        
+        return embedding
+    
+    def _generate_fallback_embedding(self, text: str) -> List[float]:
+        """Generate a simple fallback embedding when OpenAI is not available"""
+        # Create a deterministic embedding based on text hash
+        text_hash = hashlib.md5(text.encode()).hexdigest()
```

Ruff (0.12.2)

Probable use of insecure hash functions in `hashlib`: `md5`

(S324)